glsk/glsk-document-api/src/main/java/com/powsybl/glsk/api/util/converters/GlskPointLinearGlskConverter.java

```diff
@@ -127,23 +135,28 @@ private static void convertExplicitProportional(Network network, GlskShiftKey gl
                     .map(network::getGenerator)
                     .filter(NetworkUtil::isCorrect)
                     .collect(Collectors.toList());
-            double totalP = generators.stream().mapToDouble(NetworkUtil::pseudoTargetP).sum();
+            totalP += generators.stream().mapToDouble(NetworkUtil::pseudoTargetP).sum();
+            double finalTotalP = totalP;
```

Comment:
Replace foreach method on streams with standard foreach loop to avoid this strange finalTotalP variable need.
(same in many places)